lib/powerpoint/slide/content.rb

```diff
@@ -0,0 +1,224 @@
+        if !@coords
+          @coords = {}
+        else
+          @coords = boxed_coordinates(@coords, FastImage.size(@image_path), @centered)
+        end
+        @image_name = File.basename(@image_path)
+      end
+
+      def file_type
+        File.extname(image_name).gsub('.', '')
+      end
+
+      def render
+        render_str('picture.xml.erb')
+      end
+    end
+
+    class TableContent
+      include Powerpoint::Util
+
+      attr_reader :content, :id, :idx
+      attr_writer :id, :idx
+
+      def initialize(options={})
+        require_arguments [:content, :idx], options
+        options.each {|k, v| instance_variable_set("@#{k}", v)}
+
+      end
+
+      def render
+        render_str('table.xml.erb')
+      end
+    end
+
+    # Allows us to use formatting inside of strings
+    #
+    # Usage:
+    # Text.new('my string', size: 1200, color: 'ab041g') + ' has ' + Text.new('sizes', size: 1600)
+    #
+    #
+    class Text
+      include Powerpoint::Util
+
+      attr_reader :values, :next, :options
+      attr_writer :next
+
+      def initialize(values, options={})
+        # require_arguments [], options
+        options.each { |k, v| instance_variable_set("@#{k}", v) }
+        if values.instance_of? String
+          @values = [values]
+        else
+          @values = values
+        end
+
+        @options = options
+        @bullets = false unless @bullets
+      end
+
+      def + (other)
+        if other.instance_of? String
+          if @options.has_key?(:color) || @options.has_key?(:size)
+            self.copy(Text.new(other))
+          else
+            Text.new(@values + [other], options)
+          end
+        else
+          self.copy(other.copy(nil))
+        end
+      end
+
+      def copy(new_next)
+        next_opt = @options.clone
+        if next_opt[:next] != nil
+          next_opt[:next] = next_opt[:next].copy(new_next)
+        elsif new_next != nil
+          next_opt[:next] = new_next
+        end
+
+        Text.new(values, next_opt)
+      end
+
+      def to_str
+        next_str = @next == nil ? 'nil' : @next.to_str
+        "Text['#{values.join("', '")}'] ~> #{next_str}"
+      end
+
+      def render(bullets = false)
+
+        header = '<a:p>
+                    <a:pPr marL="0" indent="0">
+                      <a:spcBef>
+                        <a:spcPts val="0"/>
+                      </a:spcBef>' +
+                      (bullets ? '' : '<a:buNone/>') +
+                    '</a:pPr>'
+
+        footer = '</a:p>'
+
+        tmp = self
+        output = header
+        while tmp != nil
+          output += tmp.render_elem
+          tmp = tmp.next
+        end
+        output + footer
+      end
+
+      def render_elem
+        sz = (options[:size] == nil) ? ' ' : " sz=\"#{options[:size]}\""
+        color = (options[:color] == nil) ? '' : "<a:solidFill>
+                                                  <a:srgbClr val=\"#{options[:color]}\"/>
```

Comment:
fyi, there's [another string literal syntax](http://www.zenspider.com/Languages/Ruby/QuickRef.html#strings) I find ideal for situations like this:
```ruby
str = %(In this string, no need to escape "double quotes" or 'single quotes', but can still interpolate #{variables})
```